app/controllers/search_controller.rb

```diff
@@ -1,30 +1,30 @@
 class SearchController < ApplicationController
 
-  before_action :ensure_user_area
+  before_action :ensure_user_municipality
 
   def show
-    @demands = user_area.demands.to_a
+    @demands = user_municipality.demands.to_a
 
     if params[:q].present?
       # TODO: replace this with a proper search engine. elasticsearch?
       query = params[:q].split(' ').join('|')
 
-      demands = user_area.demands.advanced_search(solution: query)
+      demands = user_municipality.demands.advanced_search(solution: query)
 
-      demands_by_problem = user_area
+      demands_by_problem = user_municipality
                    .problems
                    .includes(:demands)
-                   .advanced_search(name: query).map { |problem| problem.demands.find { |demand| demand.area == user_area }}.compact
+                   .advanced_search(name: query).map { |problem| problem.demands.find { |demand| demand.area == user_municipality }}.compact
```

Comment:
This'll have to eventually compare the `demand.area` to the municipality, county, and state. Probably fine for now but probably won't scale.